compiler/rustc_interface/src/passes.rs

```diff
@@ -492,16 +491,22 @@ fn write_out_deps(tcx: TyCtxt<'_>, outputs: &OutputFilenames, out_filenames: &[P
 
             // Emit special comments with information about accessed environment variables.
             let env_depinfo = sess.psess.env_depinfo.borrow();
+
+            // We will soon sort, so the initial order does not matter.
+            #[allow(rustc::potential_query_instability)]
+            let mut env_depinfo: Vec<_> = env_depinfo
+                .iter()
+                .map(|(k, v)| (escape_dep_env(k.as_str()), v.map(|v| escape_dep_env(v.as_str()))))
+                .chain(tcx.sess.target.is_like_osx.then(|| {
+                    // On Apple targets, we also depend on the deployment target environment variable.
+                    let name = rustc_target::spec::apple_deployment_target_env(&tcx.sess.target.os);
+                    (name.into(), std::env::var(name).ok().map(|var| escape_dep_env(&var)))
+                }))
+                .collect();
+            env_depinfo.sort_unstable();
```

Comment:
Question [REV1]: `sort_unstable` here. Can this ever cause reproducibility problems? Or rather, is there any particular reason this needs to be `sort_unstable` instead of a stable sort?

Reply:
No idea, this is mostly existing code, I just moved it

Reply:
Thinking about it, I doubt that the ordering matters, this is environment variable names and values that are passed to Cargo, which in turn uses them all at the same time to determine whether to rebuild.